### PR TITLE
Improve disk space usage for source container builds 

### PR DESCRIPTION
**Pull request by rcerven**
Signed-off-by: Robert Cerven <rcerven@redhat.com>

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- n/a JSON/YAML configuration changes are updated in the relevant schema
- n/a Changes to metadata also update the documentation for the metadata
- n/a Pull request has a link to an osbs-docs PR for user documentation updates
- n/a New feature can be disabled from a configuration file

### CODE REVIEW

**Review by chmeliik — COMMENTED** [review]
Could you please include some more information in the commit message?

One thing that is not entirely clear to me - this does not affect the size of the final source container, does it? Only the space that is used while building that container?

File: atomic_reactor/plugins/build_source_container.py [hunk 1]
```diff
@@ -123,10 +124,34 @@ def run(self):
...
+
+        image_tar_path = self.export_image(image_output_dir)
+
+        self.log.info('Will remove unpacked image directory: %s', image_output_dir)
+        shutil.rmtree(image_output_dir)
 
         return BuildResult(
             logs=output,
-            oci_image_path=image_output_dir,
+            oci_image_path=image_tar_path,
```
[the remark this answers is not in the document]
Comment (rcerven) [reply atomic_reactor/plugins/build_source_container.py:155]: yeah I know, problem is that I'd have to remove BuildResult and maybe in other places as well ...

File: atomic_reactor/plugins/post_tag_and_push.py [hunk 2]
```diff
@@ -98,7 +98,7 @@ def push_with_skopeo(self, registry_image, insecure, docker_push_secret,
                 raise RuntimeError("Attempt to push unsupported image type %s with skopeo" %
                                    image['type'])
         else:
-            source_img = 'oci:{}'.format(source_oci_image_path)
+            source_img = 'docker-archive:{}'.format(source_oci_image_path)
```
[the remark this answers is not in the document]
Comment (rcerven) [reply atomic_reactor/plugins/post_tag_and_push.py:101]: right, in this plugin I can rename that variable

File: atomic_reactor/plugins/build_source_container.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (chmeliik) [reply atomic_reactor/plugins/build_source_container.py:155]: What do you mean remove? Could only the `oci_image_path` attribute be renamed?

[the remark this answers is not in the document]
Comment (rcerven) [reply atomic_reactor/plugins/build_source_container.py:155]: I've meant rename :) let me check it

[the remark this answers is not in the document]
Comment (rcerven) [reply atomic_reactor/plugins/build_source_container.py]: ok renamed

[the remark this answers is not in the document]
Comment (chmeliik) [reply atomic_reactor/plugins/build_source_container.py]: Thanks!

**Comment by chmeliik** [comment]
> Could you please include some more information in the commit message?

@rcerven pretty please 😊

> One thing that is not entirely clear to me - this does not affect the size of the final source container, does it? Only the space that is used while building that container?

Is my understanding correct?

**Comment by rcerven** [comment]
> > Could you please include some more information in the commit message?
> 
> @rcerven pretty please blush
sure
> 
> > One thing that is not entirely clear to me - this does not affect the size of the final source container, does it? Only the space that is used while building that container?
> 
> Is my understanding correct?

yes exactly, it isn't about image size itself, but about disk space consumption during  build

**Review by lkolacek — APPROVED** [review]

**Comment by rcerven** [comment]
> > > Could you please include some more information in the commit message?
> > 
> > 
> > @rcerven pretty please blush
> > > One thing that is not entirely clear to me - this does not affect the size of the final source container, does it? Only the space that is used while building that container?
> > 
> > 
> > Is my understanding correct?
> 
> yes exactly, it isn't about image size itself, but about disk space consumption during build
> 
> sure I'll add more info in the commit message

added more info into the commit message

**Review by chmeliik — APPROVED** [review]
LGTM 👍

**Comment by rcerven** [comment]
rn updated: Improve disk space usage for source container builds

**Pull request merged**